collector-http/src/main/java/zipkin2/collector/otel/http/OpenTelemetryHttpCollector.java

```diff
@@ -68,31 +93,86 @@ public static final class Builder extends CollectorComponent.Builder {
     metrics = builder.metrics;
   }
 
-  @Override public OpenTelemetryHttpCollector start() {
+  @Override
+  public OpenTelemetryHttpCollector start() {
     return this;
   }
 
-  @Override public String toString() {
+  @Override
+  public String toString() {
     return "OpenTelemetryHttpCollector{}";
   }
 
   /**
    * Reconfigures the service per https://opentelemetry.io/docs/specs/otlp/#otlphttp-request
    */
-  @Override public void reconfigure(ServerBuilder sb) {
+  @Override
+  public void reconfigure(ServerBuilder sb) {
     sb.service("/v1/traces", new HttpService(this));
   }
 
   static final class HttpService extends AbstractHttpService {
+
     final OpenTelemetryHttpCollector collector;
 
     HttpService(OpenTelemetryHttpCollector collector) {
       this.collector = collector;
     }
 
-    @Override protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req)
+    @Override
+    protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest req)
         throws Exception {
-      throw new RuntimeException("Implement me!");
+      CompletableCallback result = new CompletableCallback();
+      req.aggregate(AggregationOptions.usePooledObjects(ByteBufAllocator.DEFAULT, ctx.eventLoop()
+      )).handle((msg, t) -> {
+        if (t != null) {
+          result.onError(t);
+          return null;
+        }
+        try (HttpData content = msg.content()) {
+          if (content.isEmpty()) {
+            result.onSuccess(null);
+            return null;
+          }
+
+          ByteBuf byteBuf = content.byteBuf();
+          final ByteBuffer nioBuffer = byteBuf.nioBuffer();
+          try (ReadBuffer readBuffer = ReadBuffer.wrapUnsafe(nioBuffer)) {
+            try {
+              InputStream inputStream = readBuffer;
+              if ("gzip".equalsIgnoreCase(req.headers().get("content-encoding"))) {
```

Comment:
We should add `DecodingService` in `reconfigure` instead of handling content-encoding ourselves

https://github.com/line/armeria/blob/main/core/src/main/java/com/linecorp/armeria/server/encoding/DecodingService.java#L49

It doesn't support `base64` in it's default decoders list, but I've never seen that encoding before and don't see it mentioned in otlp docs (it just mentions gzip) so suspect the default is fine.

Tried to remember why we don't use `DecodingService` in zipkin-server and couldn't (looked through https://github.com/openzipkin/zipkin/pull/2348). It might be related to metrics there but I can't see any reason to avoid it here.

Reply:
Great hint, I've updated the PR